include/mbedtls/ecdh.h

```diff
@@ -40,6 +40,39 @@
 
 #include "mbedtls/ecp.h"
 
+/**
+ * \def MBEDTLS_ECDH_LEGACY_CONTEXT
+ *
+ * Use a backward compatible ECDH context.
+ *
+ * Mbed TLS supports two formats for ECDH contexts (#mbedtls_ecdh_context
+ * defined in `ecdh.h`). For most applications, the choice of format makes
+ * no difference, since all library functions can work with either format,
+ * except that the new format is incompatible with MBEDTLS_ECP_RESTARTABLE.
+
+ * The new format used when this option is disabled is smaller
+ * (56 bytes on a 32-bit platform). In future versions of the library, it
+ * will support alternative implementations of ECDH operations.
+ * The new format is incompatible with applications that access
+ * context fields directly and with restartable ECP operations.
+ *
+ * Define this macro if you enable MBEDTLS_ECP_RESTARTABLE or if you
```

Comment:
I think this paragraph should be removed, as this is no longer a user-configurable option.

Reply:
Thanks @mpg ! I wasn't quite sure how to handle the documentation of this bit so I appreciate your comments. I will fix them up and push again.

Reply:
@mpg - I think I have done what you wanted me to do, please re-review when you get the chance.